src/dependencies.jl

```diff
@@ -983,6 +983,9 @@ using SHA
 function sha_check(path, sha)
     open(path) do f
         calc_sha = sha256(f)
+        if !isa(calc_sha, @compat AbstractString)
```

Comment:
Actually, you don't even need to write `@compat` for types, which are provided as aliases.

Also, a small comment to know when this can happen would be useful. It will allow removing this hack easily in the future.